kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -171,6 +171,35 @@ private static String trimVersion(String apiVersion) {
     }
   }
 
+  /**
+   * Helper method for list() and list(limit, continue) methods
+   *
+   * @param url
+   * @return list of corresponding Kubernetes Resources
+   */
+  private L listRequestHelper(URL url) {
+    try {
+      HttpUrl.Builder requestUrlBuilder = HttpUrl.get(url).newBuilder();
+
+      String labelQueryParam = getLabelQueryParam();
+      if (Utils.isNotNullOrEmpty(labelQueryParam)) {
```

Comment:
```java
 if (Utils.isNotNullOrEmpty(labelQueryParam)) {
    requestUrlBuilder.addQueryParameter("labelSelector", labelQueryParam);
 }
```

Could extract in method like

```java
void addQSParam(String name, value) {

}
```

then call it
```java
addQSParam("labelSelector", labelQueryParam);
addQSParam("fieldSelector", fieldQueryString);
```

DRY (Dont repeat yourself)